modelling/src/main/java/org/axonframework/modelling/command/ApplyMore.java

```diff
@@ -59,4 +80,23 @@
      * @return an instance of ApplyMore to apply any subsequent events
      */
     ApplyMore andThen(Runnable runnable);
+
+    /**
+     * Conditionally execute the given {@code runnable} after applying the previous event. This {@code runnable} is
+     * guaranteed to be invoked when the previous event has been fully applied to the aggregate.
+     * <p>
+     * The given {@code runnable} must not directly alter any state of the aggregate. Instead, it should only decide
+     * if more events should be applied based on the state of the aggregate after the previous event
+     *
+     * @param condition the condition to evaluate and decide if the code should be executed
+     * @param runnable  the code to execute when the previous event was applied
+     * @return an instance of ApplyMore to apply any subsequent events
+     */
+    default ApplyMore andThenIf(Supplier<Boolean> condition, Runnable runnable) {
```

Comment:
I'd like to see a test case for this addition too.

Reply:
included in 522b44418c86c25c92727eef6858d16ca14ab052